### ISSUE TITLE
How can I open numberpad automatically?

### ISSUE THREAD

**Issue by eswarreddy43** [driftyco/ionic-plugins-keyboard#45]
First of all, thanks for this wonderful plugin. I am using this plugin in my sample phonegap application.
In `onDeviceReady()`, I am calling `cordova.plugins.Keyboard.show();` for opening the keyboard, by default it is opening a keyboard contains of both alphabets, numbers. I am using my input as `type='number'`. So when I click in to that text box, only `numberpad` is displaying. So how can i display this number keyboard by using `cordova.plugins.Keyboard.show();`. Please suggest.

**Comment by jvelo** [comment]
Calling ```.focus()``` is not enough, Android will focus the input but won't pop the keyboard. If you still need that behavior, I've written a plugin that emulates a native touch to do that : https://github.com/46cl/cordova-android-focus-plugin/

**Comment by zmandel** [comment]
you can only call focus() from within a user event (a click for example) but not at any point. Thats the whole point of this plugin, but it cant show numeric keyboards :(

**Comment by tlancina** [comment]
As far as I know, there is no way to do this from the native side of things.  Is there any reason you can't fake it?

```
var fakeClick = new MouseEvent('fakeClick', {
    'view': window,
    'bubbles': true,
    'cancelable': true
  });

document.body.addEventListener('fakeClick', function(e){
    e.target.focus();
});

document.body.querySelector('input').dispatchEvent(fakeClick);
```

**Comment by runspired** [comment]
I think the problem might have more to do with the project settings not being set to allow the keyboard to be triggered programmatically.  Your config.xml should contain

`<preference name="KeyboardDisplayRequiresUserAction" value="false" />`

I've had no trouble programmatically triggering the keyboard on iOS or Android by triggering `.focus()` on an `<input type="text" pattern="[0-9]*">`

**Comment by tlancina** [comment]
Hey @runspired I believe that's an iOS only configuration.  I can't remember, but I seem to recall some Android versions requiring user interaction (fake or otherwise) to show the keyboard on `focus()`, although testing on a 4.4 device I can also get the keyboard to show just fine without an event.

Closing this, since you can do this with JavaScript just fine as far as I can tell.

**Comment by zmandel** [comment]
Hi all. I happen to have worked on this just yesterday. Its not really true that you can bring the keyboard up with a simple focus() call. That only works if you call focus() within a handler for a user action (like a click on a button). Otherwise the focus will not bring up the keyboard.
This other plugin does display the keyboard correctly in all situations and android versions:
https://github.com/46cl/cordova-android-focus-plugin

it does so by basically faking a click (from the java plugin side) into a dom element you specify from the javascript.

**Comment by tlancina** [comment]
@zmandel what device and what version of Android are you using?

**Comment by zmandel** [comment]
on a physical 4.4 and tried on emulated 2.3 and 4.0.
setting focus to a number input outside of a user click (user action in
general) will not bring up the keyboard, it will just set the focus.

example: on a click event, call focus from a setTimeout of 1000ms. Since
the focus gets called outside of the user action, android will not bring
the keyboard up.
in my case, my app does an animation after the user clicks, and I need to
wait until the animation finishes to bring up the keyboard. if i bring it
at the same time as the animation, the animation wont be visible and/or
will be slow since the keyboard also animates.

The plugin i suggested fixes this situation. Ive tested that plugin on 2.3,
4.0 and 4.4 and it works (after I fixed a few issues which the author
already incorporated into the plugin)

Zig Mandel <https://developers.google.com/experts/people/zig-mandel>

**Comment by jvelo** [comment]
@tlancina I can confirm you can't pop the soft keyboard with focus outside an user action (my use case for instance was to show it right away within Ionic platform ready). I've tested on several devices, including a Galaxy S3 mini running Android 4.4, a Nexus 5 running 5.0.

**Comment by tlancina** [comment]
Yup can confirm this, sorry for my initial confusion.  If I tap the page _once_ then I can focus() no problem to bring up the keyboard any amount of subsequent times, but without the initial user interaction it won't show. Android is so weird sometimes.

Re-opened, @jvelo any interest in submitting a PR to merge your plugin into a `focus` method for Keyboard?

**Comment by tlancina** [comment]
Given that it seems like there only needs to be one initial user interaction, I wonder if there is some way we could "initialize" the web view when the plugin loads that would enable focus() to work from javascript.

Faking an unwanted touch could be problematic, but maybe there is some other way to tell Android that the user has interacted with it?

**Comment by jvelo** [comment]
@tlancina yes I'm interested in mergin my work in the keyboard plugin.

**Comment by jvelo** [comment]
I haven't found another way to force the interaction than this fake touch ; though I don't know very well how Android works. Maybe someone with more in-depth knowledge of the platform would know of alternatives.

**Comment by zmandel** [comment]
in my comments above I suggest looking at https://github.com/46cl/cordova-android-focus-plugin, I contributed to that one and it implements well the fake click.

**Comment by deciomoritz** [comment]
@jvelo's plugin solves everything. Using type="text" for default keyboard and type="tel" for numeric keyboard. The plugin simulates a user click. Works like a charm.
